### PR TITLE
Improve model support

### PR DESCRIPTION
**Pull request by acron0**
Also rename `defworkflowmodel` to `defmodel`

### CODE REVIEW

File: src/witan/workspace_api.clj [hunk 1]
```diff
@@ -65,9 +59,37 @@
...
+
+(defn model-valid?
+  [{:keys [workflow catalog]}]
+  (try
+    (let [node-names    (-> workflow flatten set)
+          catalog-names (set (map :witan/name catalog))
+          groups        (group-by :witan/name catalog)]
+      (cond
+        (not-empty (clojure.set/difference node-names catalog-names))
+        (do (println "An error occurred: There are missing :witan/name entries in the catalog.")
```
Comment (tcoupland) [inline src/witan/workspace_api.clj]: These do's are a little unnecessary. println with return a 'nil', which is treated as false.

Comment (tcoupland) [inline src/witan/workspace_api.clj]: also 'println's?! :)

Comment (acron0) [inline src/witan/workspace_api.clj]: Haha, yeah, this is this only obvious way to get information out of a schema constraint predicate. I shall remove the `do`s but...these will only occur at compile time (and will prevent compilation) so...acceptable?

**Comment by tcoupland** [comment]
:shipit:

**Pull request merged**